### ISSUE TITLE
Prevent authenticated users from accessing other users' profiles

### ISSUE THREAD

**Issue by AmineTissilguit** [ID-JA/2tinylink#37]
- Currently, any authenticated user can access another user's profile at `http://localhost:5132/api/Profiles/{username}` by simply knowing their username. To ensure that only users who own the username can view the associated profile, access needs to be restricted.

**Comment by ID-JA** [comment]
Fixed with #163

**Comment by ID-JA** [comment]
Fixed with #164 
To solve the issue, I removed the `\profiles\{username}` endpoint and replaced it with a new endpoint `\profiles\me`. When calling this endpoint, you will receive information about the authenticated user.

```diff
// ProfilesController.cs
using Application.UseCases.Settings.Queries;
using Application.UseCases.UserManagement.Queries.UserByUserName;
using MediatR;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using WebUI.Contracts.UserManagement.UserByUserName;

namespace WebUI.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class ProfilesController : ControllerBase
    {
        private readonly ISender _sender;

        public ProfilesController(ISender sender)
        {
            _sender = sender;
        }

-        [HttpGet("{userName}", Name = "GetUserProfileByUserName")]
-        public async Task<ActionResult<UserByUserNameResponse>> GetUserProfileByUserName([FromRoute] UserByUserNameResponse request)
-       {
-            var query = new UserByUserNameQuery { UserName = request.UserName };
-
-            var result = await _sender.Send(query);
-
-            var response = new UserByUserNameResponse()
-            {
-                FirstName = result.FirstName,
-                LastName = result.LastName,
-                UserName = result.UserName,
-                CreatedAt = result.CreatedAt
-            };
-
-           return Ok(response);
-        }

+       [HttpGet("me", Name = "GetCurrentUser")]
+        [Authorize]
+        public async Task<ActionResult<UserByUserNameResponse>> GetCurrentUser()
+       {
+          return Ok(await _sender.Send(new GetUserProfileQuery()));
+      }
+    }
}
```

I implemented an interface called `ICurrentUser` that provides methods to retrieve authenticated user information.

```cs
// ICurrentUser.cs
using System.Security.Claims;

namespace Application.Common.Interfaces;

public interface ICurrentUser
{
    string? Name { get; }

    string GetUserId();

    string GetUserEmail();

    bool IsAuthenticated();
    void SetCurrentUser(ClaimsPrincipal user);

    void SetCurrentUserId(string userId);

    IEnumerable<Claim>? GetUserClaims();

}

```
